src/Core/Repositories/SqlServer/EventRepository.cs

```diff
@@ -74,14 +74,14 @@ public async Task CreateAsync(IEvent e)
             await base.CreateAsync(ev);
         }
 
-        public async Task CreateManyAsync(IList<IEvent> entities)
+        public async Task CreateManyAsync(IEnumerable<IEvent> entities)
         {
             if (!entities?.Any() ?? true)
             {
                 return;
             }
 
-            if (entities.Count == 1)
+            if (!entities.Skip(1).Any())
```

Comment:
Perf over Count()?

Reply:
Tests show that the actual property vs methods is ~ 100x slow to use Linq. However, if you count the time to create a list vs array, it's more like a factor of 2 in favor of Linq.

BTW, in order to detect this 100x difference I ran `Count` vs `Skip(1).Any()` 10 million times. I don't think speed is an issue here, and we gain by being more lenient with the parameter type, since we're likely to be performing Linq operations on these prior to calling this Enqueue method.

Reply:
I too observed this behavior with IEnumerable a few years ago and documented it here:
https://gist.github.com/kspearrin/8dbf7d76d6fd21a8d861dc9c472547d4

Reply:
Data

**Object creation included in time**
| iterations | length | list (ms) | array (ms) |
| --------- | ------- | ------- | ---------- |
| 100k | 10k | 2417 | 1158|
| 100k | 1k | 296 | 138 |
| 100k | 100 | 40 | 22 |
| 100k | 10 | 13 | 14 |
| 100k | 1 | 9 | 7 |

**Object created before timer**

| iterations | length | list (ms) | array (ms) |
| --------- | ------- | ------- | ---------- |
| 10M | 10k | 18 | 977 |
| 10M | 1k | 18 | 827 |
| 10M | 100 | 18 | 830 |
| 10M | 10 | 21 | 824 |
| 10M | 1 | 19 | 769 |

Reply:
Yeah, you definitely need to be careful to make sure you're not iterating the list. That gist is good data for sure, thanks